android/src/main/java/com/gettipsi/stripe/dialog/EnterAddressDialogFragment.java

```diff
@@ -0,0 +1,126 @@
+package com.gettipsi.stripe.dialog;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.content.ContextCompat;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.EditText;
+
+import com.gettipsi.stripe.R;
+import com.gettipsi.stripe.util.PlainAddress;
+import com.gettipsi.stripe.util.DataUtil;
+
+public class EnterAddressDialogFragment extends DialogFragment {
+  public AddCardDialogFragment next = null;
```

Comment:
Very, very bad idea. Fragment behaviour shouldn't rely on instance members. Fragment's Instance members should be initiated on instance creation and be managed by fragment itself. Setting references from outside is direct route to the crash. Any time it can be recreated and field will be null/inconsistent. Try passing data thru explicit means like bundle or at least keep in mind Fragments lifeCycleMethods to avoid undefined behaviour. AFAIU you have to delegate AddCardDialogFragment creation to EnterAddressDialogFragment. So first you show EnterAddressDialogFragment then pass collected data via bundle and show it. Stop stop OMG. It's all unnecessary because we need to pass Promise to =( So seems nothing we really can do because promise design is already suck androidLifeCycle. Ok, at least null-check "next" field